src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,294 @@
+    private static final String BYE = "bye";
+    private static final String BY = "/by";
+    private static final String AT = "/at";
+    private static final String DELETE = "delete";
+    private static final String DOTTED_LINE = "____________________________________________________________";
+    private static final String LOGO = (" ____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n")
+            .replaceAll("\n", System.lineSeparator() + "\t");
+
+    private static final int INSTRUCTION_LENGTH = 2;
+
+    private final ArrayList<Task> tasks;
+    private int taskCount;
+
+    private boolean isLoaded;
+
+    Duke() {
+        tasks = new ArrayList<>();
+        taskCount = 0;
+        isLoaded = false;
+    }
+
+    public static void main(String[] args) {
+        Duke duke = new Duke();
+        duke.loadTasks();
+        duke.greetUser();
+        duke.processInput();
+    }
+
+    private void printWithIndent(String string) {
+        System.out.println("\t" + string);
+    }
+
+    private void loadTasks() {
+        Path filePath = Paths.get(FOLDER_PATH, FILE_NAME);
+
+        try {
+            try {
+                if (Files.exists(filePath)) {
+                    BufferedReader bufferedReader = Files.newBufferedReader(filePath);
+
+                    while (true) {
+                        String line = bufferedReader.readLine();
+                        if (line == null) {
+                            break;
+                        }
+                        String[] instructions = line.split("\\|", 3);
+                        boolean isDone = instructions[1].trim().equals("1");
+                        switch (instructions[0].trim()) {
+                        case "D":
+                            addDeadline(instructions[2], isDone);
+                            break;
+                        case "E":
+                            addEvent(instructions[2], isDone);
+                            break;
+                        case "T":
+                            addToDo(instructions[2], isDone);
+                            break;
+                        default:
+                            throw new DukeException("Error reading file!");
+                        }
+                    }
+                }
+            } catch (IOException e) {
+                throw new DukeException("Error reading file!");
+            }
+        } catch (DukeException ignored) {
+
+        }
+
+        isLoaded = true;
+
+    }
+
+    private void greetUser() {
+        printWithIndent(DOTTED_LINE);
+        printWithIndent(LOGO);
+        printWithIndent(" Hello! I'm Duke");
+        printWithIndent(" What can I do for you?");
+        printWithIndent(DOTTED_LINE);
+    }
+
+    private void deleteTask(String instruction) throws DukeException {
+
```

Comment:
Could maybe take away this leading empty line for consistency with the other methods